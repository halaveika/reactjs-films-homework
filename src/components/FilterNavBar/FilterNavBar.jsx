import React from 'react';
import { Tabs } from 'antd';
import DropDown from '../DropDown';
import PropTypes from 'prop-types';
import {TRENDING,TOPRATED,COMINGSOON,GENRE} from '../../constants';
import './FilterNavBar.scss';
const { TabPane } = Tabs;

 const FilterNavBar = ({genres, handleGenre,activeGenre,setFilter,filter,setCurrentPage}) => {
  function callback(key) {
    if (key === TRENDING) {
      setFilter(TRENDING)
      setCurrentPage(1);}
    if (key === TOPRATED) {
      setFilter(TOPRATED)
      setCurrentPage(1);}
    if (key === COMINGSOON) {
      setFilter(COMINGSOON)
      setCurrentPage(1);}
    if (key === GENRE) {
      setCurrentPage(1);}
  }
  return (
    <div className="filter-navbar">
      <Tabs defaultActiveKey={filter} onChange={callback}>
        <TabPane tab="Trending" key={TRENDING}>
        </TabPane>
        <TabPane tab="Top Rated" key={TOPRATED}>
        </TabPane>
        <TabPane tab="Coming Soon" key={COMINGSOON}>
        </TabPane>
        <TabPane tab={ <DropDown handleGenre={handleGenre} genres={genres}/> } key={GENRE}>
        </TabPane>
      </Tabs>
        <span className="genre-status">{activeGenre}</span>
    </div>
  );
}


FilterNavBar.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string})),
  handleGenre: PropTypes.func.isRequired,
  activeGenre: PropTypes.string.isRequired,
  setFilter: PropTypes.func.isRequired,
  filter: PropTypes.string.isRequired,
  setCurrentPage: PropTypes.func.isRequired
};

export default FilterNavBar;
