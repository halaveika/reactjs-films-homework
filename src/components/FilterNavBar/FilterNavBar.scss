.ant-tabs-top {
  height: 46px;
}

.filter-navbar {
  width: 100%;
}

.menu-item {
  color: #1890ff;
  cursor: pointer;

  &:hover,
  &:active {
    text-decoration: underline;
    font-weight: bold;
  }
}

.genre-status {
  position: absolute;
  top: 10px;
  left: 440px;
  background: #1890ff;
  color: #fff;
  padding: 0 1px;
  border-radius: 25%;
}

.ant-tabs-nav-wrap {
  width: 100%;
}

.ant-tabs-nav-list {
  width: 100%;
  display: flex;
  justify-content: flex-start;
}
