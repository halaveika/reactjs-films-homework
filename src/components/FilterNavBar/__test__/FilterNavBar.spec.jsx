import React from 'react';
import FilterNavBar from '..';
import { render } from "react-dom";
import { act } from "react-dom/test-utils";

const mockFilterNavBarProps = {
  genres: [{
    id: 13,
    name: 'Adventure',
  }, {
    id: 18,
    name: 'Drama',
  }, {
    id: 20,
    name: 'Family',
  }, {
    id: 21,
    name: 'Fantasy',
  }],
  filter: 'TRENDING',
  getContent: jest.fn(),
  GetGenres: jest.fn(),
  setFilter: jest.fn(),
  setCurrentPage: jest.fn(),
  setGenre: jest.fn(),
};

describe('test FilterNavBar component', () => {
  it("FilterNavBar changes value when clicked TRENDING", async() => {
  let container = null;
  container = document.createElement("div");
  document.body.appendChild(container);
  act(() => {
    render(
      <FilterNavBar
        genres={mockFilterNavBarProps.genres}
        setGenre={mockFilterNavBarProps.setGenre}
        activeGenre={mockFilterNavBarProps.activeGenre}
        setFilter={mockFilterNavBarProps.setFilter}
        filter={mockFilterNavBarProps.filter}
        setCurrentPage={mockFilterNavBarProps.setCurrentPage}
      />, container);
  });

  const button = document.querySelectorAll(".ant-tabs-tab-btn")[0];
  await act(async() => {
    button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
  });
  expect(container).toMatchSnapshot();
  
  });

  it("FilterNavBar changes value when clicked TOPRATED", async() => {
    let container = null;
    container = document.createElement("div");
    document.body.appendChild(container);
    act(() => {
      render(
        <FilterNavBar
          genres={mockFilterNavBarProps.genres}
          setGenre={mockFilterNavBarProps.setGenre}
          activeGenre={mockFilterNavBarProps.activeGenre}
          setFilter={mockFilterNavBarProps.setFilter}
          filter={mockFilterNavBarProps.filter}
          setCurrentPage={mockFilterNavBarProps.setCurrentPage}
        />, container);
    });
  
    const button = document.querySelectorAll(".ant-tabs-tab-btn")[1];
    await act(async() => {
      button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    expect(container).toMatchSnapshot();      
  });

  it("FilterNavBar changes value when clicked COMINGSOON", async() => {
    let container = null;
    container = document.createElement("div");
    document.body.appendChild(container);
    act(() => {
      render(
        <FilterNavBar
          genres={mockFilterNavBarProps.genres}
          setGenre={mockFilterNavBarProps.setGenre}
          activeGenre={mockFilterNavBarProps.activeGenre}
          setFilter={mockFilterNavBarProps.setFilter}
          filter={mockFilterNavBarProps.filter}
          setCurrentPage={mockFilterNavBarProps.setCurrentPage}
        />, container);
    });
  
    const button = document.querySelectorAll(".ant-tabs-tab-btn")[2];
    await act(async() => {
      button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    expect(container).toMatchSnapshot();      
  });

  it("FilterNavBar changes value when clicked GENRE", async() => {
    let container = null;
    container = document.createElement("div");
    document.body.appendChild(container);
    act(() => {
      render(
        <FilterNavBar
          genres={mockFilterNavBarProps.genres}
          setGenre={mockFilterNavBarProps.setGenre}
          activeGenre={mockFilterNavBarProps.activeGenre}
          setFilter={mockFilterNavBarProps.setFilter}
          filter={mockFilterNavBarProps.filter}
          setCurrentPage={mockFilterNavBarProps.setCurrentPage}
        />, container);
    });
  
    const button = document.querySelectorAll(".ant-tabs-tab-btn")[3];
    await act(async() => {
      button.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });
    expect(container).toMatchSnapshot();      
  });

});
