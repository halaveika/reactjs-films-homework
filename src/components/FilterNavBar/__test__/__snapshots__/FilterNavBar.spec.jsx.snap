// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test FilterNavBar component FilterNavBar changes value when clicked COMINGSOON 1`] = `
<div>
  <div
    class="filter-navbar"
  >
    <div
      class="ant-tabs ant-tabs-top"
    >
      <div
        class="ant-tabs-nav"
        role="tablist"
      >
        <div
          class="ant-tabs-nav-wrap"
        >
          <div
            class="ant-tabs-nav-list"
            style="transform: translate(0px, 0px);"
          >
            <div
              class="ant-tabs-tab ant-tabs-tab-active"
            >
              <div
                aria-controls="rc-tabs-test-panel-TRENDING"
                aria-selected="true"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-TRENDING"
                role="tab"
                tabindex="0"
              >
                Trending
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-TOPRATED"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-TOPRATED"
                role="tab"
                tabindex="0"
              >
                Top Rated
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-COMINGSOON"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-COMINGSOON"
                role="tab"
                tabindex="0"
              >
                Coming Soon
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-GENRE"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-GENRE"
                role="tab"
                tabindex="0"
              >
                <span
                  class="ant-dropdown-trigger ant-dropdown-link"
                  placement="bottomCenter"
                >
                  Genre 
                  <span
                    aria-label="down"
                    class="anticon anticon-down"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
                <div />
              </div>
            </div>
            <div
              class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
        <div
          class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
        >
          <button
            aria-controls="rc-tabs-test-more-popup"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-hidden="true"
            class="ant-tabs-nav-more"
            id="rc-tabs-test-more"
            style="visibility: hidden; order: 1;"
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="ant-tabs-content-holder"
      >
        <div
          class="ant-tabs-content ant-tabs-content-top"
        >
          <div
            aria-hidden="false"
            aria-labelledby="rc-tabs-test-tab-TRENDING"
            class="ant-tabs-tabpane ant-tabs-tabpane-active"
            id="rc-tabs-test-panel-TRENDING"
            role="tabpanel"
            tabindex="0"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-TOPRATED"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-TOPRATED"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-COMINGSOON"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-COMINGSOON"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-GENRE"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-GENRE"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
    <span
      class="genre-status"
    />
  </div>
</div>
`;

exports[`test FilterNavBar component FilterNavBar changes value when clicked GENRE 1`] = `
<div>
  <div
    class="filter-navbar"
  >
    <div
      class="ant-tabs ant-tabs-top"
    >
      <div
        class="ant-tabs-nav"
        role="tablist"
      >
        <div
          class="ant-tabs-nav-wrap"
        >
          <div
            class="ant-tabs-nav-list"
            style="transform: translate(0px, 0px);"
          >
            <div
              class="ant-tabs-tab ant-tabs-tab-active"
            >
              <div
                aria-controls="rc-tabs-test-panel-TRENDING"
                aria-selected="true"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-TRENDING"
                role="tab"
                tabindex="0"
              >
                Trending
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-TOPRATED"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-TOPRATED"
                role="tab"
                tabindex="0"
              >
                Top Rated
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-COMINGSOON"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-COMINGSOON"
                role="tab"
                tabindex="0"
              >
                Coming Soon
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-GENRE"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-GENRE"
                role="tab"
                tabindex="0"
              >
                <span
                  class="ant-dropdown-trigger ant-dropdown-link"
                  placement="bottomCenter"
                >
                  Genre 
                  <span
                    aria-label="down"
                    class="anticon anticon-down"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
                <div />
              </div>
            </div>
            <div
              class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
        <div
          class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
        >
          <button
            aria-controls="rc-tabs-test-more-popup"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-hidden="true"
            class="ant-tabs-nav-more"
            id="rc-tabs-test-more"
            style="visibility: hidden; order: 1;"
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="ant-tabs-content-holder"
      >
        <div
          class="ant-tabs-content ant-tabs-content-top"
        >
          <div
            aria-hidden="false"
            aria-labelledby="rc-tabs-test-tab-TRENDING"
            class="ant-tabs-tabpane ant-tabs-tabpane-active"
            id="rc-tabs-test-panel-TRENDING"
            role="tabpanel"
            tabindex="0"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-TOPRATED"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-TOPRATED"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-COMINGSOON"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-COMINGSOON"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-GENRE"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-GENRE"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
    <span
      class="genre-status"
    />
  </div>
</div>
`;

exports[`test FilterNavBar component FilterNavBar changes value when clicked TOPRATED 1`] = `
<div>
  <div
    class="filter-navbar"
  >
    <div
      class="ant-tabs ant-tabs-top"
    >
      <div
        class="ant-tabs-nav"
        role="tablist"
      >
        <div
          class="ant-tabs-nav-wrap"
        >
          <div
            class="ant-tabs-nav-list"
            style="transform: translate(0px, 0px);"
          >
            <div
              class="ant-tabs-tab ant-tabs-tab-active"
            >
              <div
                aria-controls="rc-tabs-test-panel-TRENDING"
                aria-selected="true"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-TRENDING"
                role="tab"
                tabindex="0"
              >
                Trending
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-TOPRATED"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-TOPRATED"
                role="tab"
                tabindex="0"
              >
                Top Rated
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-COMINGSOON"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-COMINGSOON"
                role="tab"
                tabindex="0"
              >
                Coming Soon
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-GENRE"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-GENRE"
                role="tab"
                tabindex="0"
              >
                <span
                  class="ant-dropdown-trigger ant-dropdown-link"
                  placement="bottomCenter"
                >
                  Genre 
                  <span
                    aria-label="down"
                    class="anticon anticon-down"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
                <div />
              </div>
            </div>
            <div
              class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
        <div
          class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
        >
          <button
            aria-controls="rc-tabs-test-more-popup"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-hidden="true"
            class="ant-tabs-nav-more"
            id="rc-tabs-test-more"
            style="visibility: hidden; order: 1;"
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="ant-tabs-content-holder"
      >
        <div
          class="ant-tabs-content ant-tabs-content-top"
        >
          <div
            aria-hidden="false"
            aria-labelledby="rc-tabs-test-tab-TRENDING"
            class="ant-tabs-tabpane ant-tabs-tabpane-active"
            id="rc-tabs-test-panel-TRENDING"
            role="tabpanel"
            tabindex="0"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-TOPRATED"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-TOPRATED"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-COMINGSOON"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-COMINGSOON"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-GENRE"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-GENRE"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
    <span
      class="genre-status"
    />
  </div>
</div>
`;

exports[`test FilterNavBar component FilterNavBar changes value when clicked TRENDING 1`] = `
<div>
  <div
    class="filter-navbar"
  >
    <div
      class="ant-tabs ant-tabs-top"
    >
      <div
        class="ant-tabs-nav"
        role="tablist"
      >
        <div
          class="ant-tabs-nav-wrap"
        >
          <div
            class="ant-tabs-nav-list"
            style="transform: translate(0px, 0px);"
          >
            <div
              class="ant-tabs-tab ant-tabs-tab-active"
            >
              <div
                aria-controls="rc-tabs-test-panel-TRENDING"
                aria-selected="true"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-TRENDING"
                role="tab"
                tabindex="0"
              >
                Trending
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-TOPRATED"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-TOPRATED"
                role="tab"
                tabindex="0"
              >
                Top Rated
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-COMINGSOON"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-COMINGSOON"
                role="tab"
                tabindex="0"
              >
                Coming Soon
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-GENRE"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-GENRE"
                role="tab"
                tabindex="0"
              >
                <span
                  class="ant-dropdown-trigger ant-dropdown-link"
                  placement="bottomCenter"
                >
                  Genre 
                  <span
                    aria-label="down"
                    class="anticon anticon-down"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
                <div />
              </div>
            </div>
            <div
              class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
            />
          </div>
        </div>
        <div
          class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
        >
          <button
            aria-controls="rc-tabs-test-more-popup"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-hidden="true"
            class="ant-tabs-nav-more"
            id="rc-tabs-test-more"
            style="visibility: hidden; order: 1;"
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="ant-tabs-content-holder"
      >
        <div
          class="ant-tabs-content ant-tabs-content-top"
        >
          <div
            aria-hidden="false"
            aria-labelledby="rc-tabs-test-tab-TRENDING"
            class="ant-tabs-tabpane ant-tabs-tabpane-active"
            id="rc-tabs-test-panel-TRENDING"
            role="tabpanel"
            tabindex="0"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-TOPRATED"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-TOPRATED"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-COMINGSOON"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-COMINGSOON"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-GENRE"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-GENRE"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
    <span
      class="genre-status"
    />
  </div>
</div>
`;
